README.md

```diff
@@ -1,7 +1,128 @@
+objects (layers, metrics, etc) that are too specialized to be added to core Keras, but that receive 
+the same level of polish and backwards compatibility guarantees as the rest of the Keras API and that 
+are maintained by the Keras team itself (unlike TFAddons).
 
-## Under Construction
-KerasCV is a relatively new project.  Expect regular updates and changes in the near future.
+# No Goals
+- **KerasCV is not a repository of blackbox end-to-end solutions (like TF Hub or ModelGarden).**
+
+    For the time being, it is focused on modular and reusable building blocks. We expect models in 
+    ModelGarden or TF Hub to eventually be built on top of the building blocks provided by KerasCV.
+
+    In the process of developing these building blocks, we will by necessity implement end-to-end 
+    workflows, but they're intended purely for demonstration and grounding purposes, they're not 
+    our main deliverable.
+
+
+- **KerasCV is not a repository of low-level image-processing ops, like tf.vision.** 
+
+    KerasCV is fundamentally an extension of the Keras API: it hosts Keras objects, like layers, 
+    metrics, or callbacks. Low-level C++ ops should go elsewhere.
+
+
+- **KerasCV is not a research library.**
+
+    Researchers may use it, but we do not consider researchers to be our target audience. Our target 
+    audience is applied CV engineers with experimentation and production needs. KerasCV should make 
+    it possible to quickly reimplement industry-strength versions of the latest generation of 
+    architectures produced by researchers, but we don't expect the research effort itself to be built
+    on top of KerasCV. This enables us to focus on usability and API standardization, and produce 
+    objects that have a longer lifespan than the average research project.
+
+## Areas of focus
+At the start of the project, we would like to take a task-oriented approach to outline our project, 
+and the first 2 tasks we would like to focus are:
+
+1. **Image classification** (Given an image and a list of labels, return the label and confidence for 
+   the image).
+2. **Object detection** (Given an image and a list of labels, return a list of bounding box, 
+   corresponding labels and confidence for the objects in the image).
+
+For each of the tasks, we are targeting to provide user model architectures, layers, preprocessing
+functions, losses, metrics and other necessary components to achieve state of the art accuracy.
+
+There will be common shareable components between those 2 tasks, like model building blocks, etc. We
+will have them in the repository as public API as well.
+
+### Common components
+- Model building block (served as backbone for the application)
+  - Resnet 
+  - Resnet RS
+  - ResNeXT
+  - Efficient
+- Preprocessing layers
+  - CutMix
+  - MixUp
+  - CutOut
+  - RandAugment
+  - RandomErasing
+
+### Image Classification
+- Dataset to evaluate
```

Comment:
For this point and the next one are we going to integrate with https://github.com/tensorflow/datasets ?

As historically in Keras we have inherited something the was create before TF datasets in the ecosystem:
https://keras.io/api/datasets/

I know that the two are currently  not strictly competing  as `api/dataset` has a very small coverage but as we plan to expand on multiple datasets it could be interesting to clarify this to the community.

Reply:
The benchmarks serve to ensure we have the components in the repository required to achieve these scores.   There's not a specific component for datasets in Keras-CV.  

Basically, we should be able to score state of the art on imagenet, regardless of source, using KerasCV components.

Reply:
But you still need to feed a dataset to produce and store the weighs and reproduce the scores (implementation reproducibility check). I suppose that in the `keras.application` this process was totally private if I remember correctly. 
Probably here we are going to store score reproducibility protocols/scripts/markdowns in this repo and just using private training job with internal resources to produce the trained weights.
But at the same time if we need to expand over https://keras.io/api/datasets/ I think this will be another source of duplication with the Tensorflow datasets repo that is trying to call the community to collect many research datasets adapters also in the Vision domain.

Reply:
> But you still need to feed a dataset to produce and store the weighs and reproduce the scores (implementation reproducibility check). I suppose that in the keras.application this process was totally private if I remember correctly.
Probably here we are going to store score reproducibility protocols/scripts/markdowns in this repo and just using private training job with internal resources to produce the trained weights.
But at the same time if we need to expand over https://keras.io/api/datasets/ I think this will be another source of duplication with the Tensorflow datasets repo that is trying to call the community to collect many research datasets adapters also in the Vision domain.

Sure, from that perspective our training scripts will probably use tensorflow datasets.  As of now, we will not be hosting datasets in this repo.

Reply:
I think we will prefer the tf.datasets as the main integration point for consuming data, and not speed much effort on expand the support for keras.datasets. The new application model in keras_cv will have tf.datasets in mind, eg it will expect the tf.dataset instance as the preferred input (whereas the keras.datasets usually produce raw numpy array)

As for the training logic and scripts, I think we should make it available for end user as well (if they want to train the model by themselves). It will not be a public API, and we will use that to produce weights/validate performance.

Reply:
Yes I agree with your point. 
Now that it is clear I think that we could add something in this section.